cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2709,6 +2743,11 @@ private List<Field<?>> getRunRecordApplicationRefPrefix(String status,
   }
 
   private List<Field<?>> getRunRecordProgramPrefix(String status, @Nullable ProgramId programId) {
+    return getRunRecordProgramPrefix(status, programId, false);
+  }
+
+  private List<Field<?>> getRunRecordProgramPrefix(String status,
+      @Nullable ProgramId programId, boolean ignoreVersions) {
```

Comment:
Just use getRunRecordProgramRefPrefix

Reply:
Used the `getRunRecordProgramRefPrefix`, also refactored it.